src/storage/MonitoringStore.ts

```diff
@@ -29,31 +30,38 @@ export class MonitoringStore<T extends ResourceStore = ResourceStore>
   }
 
   public async addResource(container: ResourceIdentifier, representation: Representation,
-    conditions?: Conditions): Promise<ResourceIdentifier> {
+    conditions?: Conditions): Promise<ModifiedResource> {
     const identifier = await this.source.addResource(container, representation, conditions);
-    this.emitChanged([ container, identifier ]);
+    this.emitChanged([ changedResource(container), identifier ]);
     return identifier;
   }
 
   public async deleteResource(identifier: ResourceIdentifier,
-    conditions?: Conditions): Promise<ResourceIdentifier[]> {
+    conditions?: Conditions): Promise<ModifiedResource[]> {
     return this.emitChanged(await this.source.deleteResource(identifier, conditions));
   }
 
   public async setRepresentation(identifier: ResourceIdentifier, representation: Representation,
-    conditions?: Conditions): Promise<ResourceIdentifier[]> {
+    conditions?: Conditions): Promise<ModifiedResource[]> {
     return this.emitChanged(await this.source.setRepresentation(identifier, representation, conditions));
   }
 
   public async modifyResource(identifier: ResourceIdentifier, patch: Patch,
-    conditions?: Conditions): Promise<ResourceIdentifier[]> {
+    conditions?: Conditions): Promise<ModifiedResource[]> {
     return this.emitChanged(await this.source.modifyResource(identifier, patch, conditions));
   }
 
-  private emitChanged(identifiers: ResourceIdentifier[]): typeof identifiers {
-    for (const identifier of identifiers) {
-      this.emit('changed', identifier);
+  private emitChanged(modified: ModifiedResource[]): ModifiedResource[] {
+    // Don't emit 'changed' event for internal resources
+    if (!this.isInternalResource(modified)) {
+      this.emit('changed', modified);
     }
-    return identifiers;
+    return modified;
+  }
+
+  private isInternalResource(result: ModifiedResource[]): boolean {
+    return result.filter(
+      (modified: ModifiedResource): boolean => modified.resource.path.includes('/.internal'),
```

Comment:
That knowledge should not leak here (and the implementation is not correct; `http://example.org/foo/.internalstuff` is not internal).

Reply:
Right 😅 Well I suppose we'll delete this `isInternalResource` function anyway since it's not the concern of this particular class to filter such(`/.internal`) things.